shared/common-adapters/tabs.tsx

```diff
@@ -1,70 +1,128 @@
 import * as React from 'react'
 import * as Styles from '../styles'
-import Box from './box'
+import Badge from './badge'
 import ClickableBox from './clickable-box'
 import Divider from './divider'
+import Icon, {IconType} from './icon'
+import ProgressIndicator from './progress-indicator'
+import Text from './text'
+import {Box, Box2} from './box'
+import capitalize from 'lodash/capitalize'
 
 const Kb = {
+  Badge,
   Box,
+  Box2,
   ClickableBox,
   Divider,
+  Icon,
+  ProgressIndicator,
+  Text,
 }
 
-type Props = {
-  clickableBoxStyle?: any
-  tabs: Array<React.ReactNode>
-  selected: React.ReactNode
-  onSelect: (s: React.ReactNode, idx: number) => any
-  style?: any
-  tabStyle?: any
+export type Tab<TitleT extends string> = {
+  title: TitleT
+  text?: string // text to show instead of title
+  icon?: IconType
+  badgeNumber?: number
 }
 
-const Tabs = ({clickableBoxStyle, tabs, selected, onSelect, style, tabStyle}: Props) => {
+type Props<TitleT extends string> = {
+  clickableBoxStyle?: Styles.StylesCrossPlatform
+  tabs: Array<Tab<TitleT>>
+  onSelect: (title: TitleT, idx: number) => void
```

Comment:
We never actually use `idx` any more, so you can just make this `onSelect: (title: TitleT)`